cachi2/core/package_managers/pip.py

```diff
@@ -134,14 +134,18 @@ def should_download(self) -> bool:
         Otherwise, we do.
         """
         return (
-            len(self.checksums_to_verify) > 0
+            len(self.checksums_to_match) > 0
             or len(self.pypi_checksums) == 0
             or len(self.user_checksums) == 0
         )
 
-    def should_verify_checksums(self) -> bool:
-        """Check if checksum verification is required."""
-        return len(self.checksums_to_verify) > 0
+    def has_checksums_to_match(self) -> bool:
```

Comment:
A very minor nitpick: this method does not accept any arguments other than `self`, it might make sense to mark it as a property.